src/intrinsics/prepack/global.js

```diff
@@ -264,13 +264,29 @@ export default function(realm: Realm): void {
         // casting to any to avoid Flow bug "*** Recursion limit exceeded ***"
         if ((object: any) instanceof AbstractObjectValue || (object: any) instanceof ObjectValue) {
           let generator = realm.generator;
-          if (generator)
-            generator.emitInvariant(
-              [object, value, object],
-              ([objectNode, valueNode]) =>
-                t.binaryExpression("!==", t.memberExpression(objectNode, t.identifier(key)), valueNode),
-              objnode => t.memberExpression(objnode, t.identifier(key))
-            );
+          if (generator) {
+            let condition = ([objectNode, valueNode]) =>
+              t.binaryExpression("!==", t.memberExpression(objectNode, t.identifier(key)), valueNode);
+            if (invariantOptions && invariantOptions instanceof StringValue) {
```

Comment:
Don't just try to get the string out manually. If `invariantOptions` is defined, call `ToStringPartial` to get the string. This will throw (and it should!) if a bogus non-string value is passed.